src/ts/background/index.ts

```diff
@@ -65,14 +69,17 @@ browser.commands.onCommand.addListener(async (command: string) => {
 
 // When tab activated, ask content script to check if the page has API documentation
 browser.tabs.onActivated.addListener(async activeInfo => {
+  let isApi = false;
   try {
-    await messageSender.sendMessageToTab(activeInfo.tabId, {
+    isApi = await messageSender.sendMessageToTab(activeInfo.tabId, {
```

Comment:
I understand the `messageSender` is the main means to communicate information. I think the different types of messages could be abstracted away as their own functions though.
`messageSender.checkIfApi(activeInfo.tabId)` or even outside of `messageSender`. It could make things clearer and simpler to use.

Reply:
👍 I did a similar thing with the strongly typed `MessageGeneric`, could of course refactor like that as well.